tests/test_tube_seg.py

```diff
@@ -229,3 +230,39 @@ def test_tubes_seg():
     assert np.unique(labels).all() in [0, 1]
     assert d_bg > radius ** 2
     assert d_tube <= radius ** 2
+
+
+def test_tubes_from_paths_bad_inputs():
+    """Tests that the tubes_from_paths method raises errors when given bad inputs.
+    """
+    sess = NeuroglancerSession(URL, 0, URL_SEG)
+    img, bbox, verts = sess.pull_voxel(2, 300, radius=5)  # A valid bbox with data.
+    G = sess.get_segments(2, bbox)
+    bbox = bbox.to_list()
+    paths = graph_to_paths(G)  # valid paths
+    size = np.subtract(bbox[3:], bbox[:3])
+    with pytest.raises(TypeError):
+        tube_seg.tubes_from_paths("asdf", paths)
+    with pytest.raises(ValueError):
+        tube_seg.tubes_from_paths((-1, -1, -1), paths)
+    with pytest.raises(TypeError):
+        tube_seg.tubes_from_paths(size, "asdf")
+    with pytest.raises(TypeError):
+        tube_seg.tubes_from_paths(size, [[0, 0, "asdf"]])
+    with pytest.raises(TypeError):
+        tube_seg.tubes_from_paths(size, paths, radius="asdf")
+    with pytest.raises(ValueError):
+        tube_seg.tubes_from_paths(size, paths, radius=-1)
+
+
```

Comment:
I see you have mainly tested valid inputs/outputs. I also like to test the actual computation on easy examples. For example, make an small image (like 10x10x10), then place two points (say, at (5,5,0) and (5,5,10)), then "draw a tube" between them (say with radius 0), and verify that the image has 1's in the appropriate locations, and 0's elsewhere

Reply:
see [here](https://github.com/neurodata/brainlit/blob/tube-connections/tests/test_tube_seg.py#L258) and [here](https://github.com/neurodata/brainlit/blob/tube-connections/tests/test_session.py#L277)